packages/client/mutations/handlers/handleJiraCreateIssue.ts

```diff
@@ -0,0 +1,49 @@
+import {ConnectionHandler, RecordProxy, RecordSourceSelectorProxy} from 'relay-runtime'
+import createProxyRecord from '~/utils/relay/createProxyRecord'
+import toJiraSearchQueryId from '~/utils/relay/toJiraSearchQueryId'
+import toTeamMemberId from '~/utils/relay/toTeamMemberId'
+import getJiraIssuesConn from '../connections/getJiraIssuesConn'
+
+const handleJiraCreateIssue = (payload: RecordProxy<any>, store: RecordSourceSelectorProxy) => {
+  const teamId = payload.getValue('teamId')
+  const meetingId = payload.getValue('meetingId')
+  const viewer = store.getRoot().getLinkedRecord('viewer')
+  const viewerId = viewer?.getValue('id') as string
+  if (!viewerId) return
+  const teamMemberId = toTeamMemberId(teamId, viewerId)
+  const teamMember = store.get(teamMemberId)
+  const integrations = teamMember?.getLinkedRecord('integrations')
+  const atlassian = integrations?.getLinkedRecord('atlassian')
+  const jiraSearchQueryId = toJiraSearchQueryId(meetingId)
+  const jiraSearchQuery = store.get(jiraSearchQueryId)
+  const queryString = jiraSearchQuery?.getValue('queryString') as string | undefined
+  const isJql = jiraSearchQuery?.getValue('isJql') as boolean | undefined
+  const projectKeyFilters = jiraSearchQuery?.getValue('projectKeyFilters') as string[] | undefined
+
+  const jiraIssue = payload.getLinkedRecord('jiraIssue')
+  const key = jiraIssue.getValue('key')
+  const summary = jiraIssue.getValue('summary')
+  const url = jiraIssue.getValue('url')
+  const newJiraIssue = {
+    key,
+    summary,
+    url
+  }
+  const jiraIssueProxy = createProxyRecord(store, 'JiraIssue', newJiraIssue)
+  const jiraIssuesConn = getJiraIssuesConn(atlassian, isJql, queryString, projectKeyFilters)
+  if (!jiraIssuesConn) return
+  const now = new Date().toISOString()
+  const newEdge = ConnectionHandler.createEdge(
+    store,
+    jiraIssuesConn,
+    jiraIssueProxy,
+    'JiraIssueEdge'
+  )
+  newEdge.setValue(now, 'cursor')
+  const node = newEdge.getLinkedRecord('node')
+  const newJiraIssueId = jiraIssue.getValue('id')
```

Comment:
Strangely, if I add `id: newJiraIssueId` to the `newJiraIssue`, I get the following error: `Cannot create a record with idffdd81d9-58c8-43af-8a23-2f6d1181cbd0:TES-315, this record already exists.` Assigning it here feels a bit hacky but this is the only way I've found of getting the correct `id` so that `UpdatePokerScopeMutation` can fire and tick the checkbox

Reply:
i should do a better job of documenting this pattern.
createProxyRecord should only be called inside an optimisticUpdater.
is there a reason you can't just use `const jiraIssue = payload.getLinkedRecord('jiraIssue')` instead of extracting out all the values & then creating a new proxy?